### PR TITLE
fix(client): Remove /force_auth query params upon successful signin.

### LINKED ISSUE #2071: clear '?email=somename@example.com' query string after force_auth signin completes
Not sure if this is really a bug, or just some oddity, but ...
                                                                                                                                            
STR

1. Sign in as a known account 'somename@example.com'
2. Load the url /force_auth?email=somename@example.com and re-enter the password
                                                                                                                                            
Expected: after login url changes to '/settings'
                            
Actual: after login url changes to '/settings?somename@example.com'.

### PR DESCRIPTION
**Pull request by shane-tomlinson**
I took this as an opportunity to do a bit of force_auth test cleanup as well.

fixes #2071 

@philbooth - Since you did all the groundwork, mind reviewing?

### CODE REVIEW

File: app/tests/spec/views/force_auth.js [hunk 1]
```diff
@@ -86,44 +87,112 @@ function (chai, $, sinon, View, FxaClient, p, AuthErrors, Relier, Broker,
...
-        assert.equal(view.$('input[type=password]').val(), 'password');
+          return view.onSignInSuccess(account);
+        });
+
+        it('invokes `afterForceAuth` on the broker', function () {
+          assert.isTrue(broker.afterForceAuth.calledWith(account));
+        });
+
+        it('navigates to the `settings` page and clears the query parameters', function () {
+          assert.isTrue(view.navigate.calledWith('settings', { clearQueryParams: true }));
```
Comment (philbooth) [inline app/tests/spec/views/force_auth.js]: Unless I'm missing it, I don't see an equivalent test for `/confirm_reset_password`. Can we add one or is that tricky for some reason?

**Comment by philbooth** [comment]
One question about the tests, but otherwise this looks good to me. :+1:

File: app/tests/spec/views/force_auth.js (hunk 1 above)
Comment (shane-tomlinson) [inline app/tests/spec/views/force_auth.js]: You are right, I totally missed it!

File: app/tests/spec/views/force_auth.js [hunk 2]
```diff
@@ -86,44 +87,112 @@ function (chai, $, sinon, View, FxaClient, p, AuthErrors, Relier, Broker,
...
+        it('user cannot create an account', function () {
+          assert.equal(view.$('a[href="/signup"]').length, 0);
+        });
+      });
+
+      describe('submit', function () {
+        var password;
+
+        beforeEach(function () {
+          password = 'password';
```
Comment (shane-tomlinson) [inline app/tests/spec/views/force_auth.js]: Move this initialization to line 109, it doesn't need to be reset each time.

**Pull request merged**